Unit test behavior influenced by external config

I'm getting an error with the unit tests when env var `SCITRAN_AUTH_CONFIG_FILE` is set with valid google, ldap and wechat sections. Unit test behavior should be isolated from such external influence.


```
Running pylint ...
=================================================== test session starts ====================================================
platform linux2 -- Python 2.7.6, pytest-2.8.5, py-1.4.33, pluggy-0.3.1
rootdir: /var/scitran/code/api/test/unit_tests/python, inifile: 
plugins: cov-2.2.0
collected 31 items 

test/unit_tests/python/test_auth.py F..
test/unit_tests/python/test_db_upgrade.py .....
test/unit_tests/python/test_files.py .....
test/unit_tests/python/test_key.py .
test/unit_tests/python/test_request.py ..
test/unit_tests/python/test_rules.py ..........
test/unit_tests/python/test_util.py ....
test/unit_tests/python/test_validators.py .

========================================================= FAILURES =========================================================
______________________________________________________ test_jwt_auth _______________________________________________________

config = AttrDict({'queue': {'max_retries': 3, 'retry_on_fail': False, 'prefetch': Fals...host': 'flywheel-elasticsearch:9200', 'db_server_selection_timeout': '20000'}})
as_drone = <conftest.ApiAccessor object at 0x7faa89700b50>, as_public = <conftest.ApiAccessor object at 0x7faa89700b10>
api_db = Database(mongomock.MongoClient('mongodb://scitran-mongo:27017/scitran-integration?replicaSet=elasticflywheel', 27017), 'scitran-integration')

    def test_jwt_auth(config, as_drone, as_public, api_db):
        # try to login w/ unconfigured auth provider
        r = as_public.post('/login', json={'auth_type': 'ldap', 'code': 'test'})
>       assert r.status_code == 400
E       assert 401 == 400
E        +  where 401 = <Response at 0x7faa86a474d0 401 Unauthorized>.status_code

test/unit_tests/python/test_auth.py:11: AssertionError
-------------------------------------------------- Captured stderr setup ---------------------------------------------------
2017-05-17 20:43:48      scitran.api                config.py   117:DEBU Database(mongomock.MongoClient('mongodb://scitran-mongo:27017/scitran-integration?replicaSet=elasticflywheel', 27017), 'scitran-integration')
2017-05-17 20:43:48      scitran.api                config.py   126:DEBU Database(mongomock.MongoClient('mongodb://scitran-mongo:27017/logs-integration?replicaSet=elasticflywheel', 27017), 'logs-integration')
--------------------------------------------------- Captured stderr call ---------------------------------------------------
2017-05-17 20:43:49      scitran.api                  base.py    63:INFO Initialized request request_id=4f622ea6-1495053829
2017-05-17 20:43:49      scitran.api                config.py   222:INFO Initializing database, creating indexes
2017-05-17 20:43:49      scitran.api                config.py   254:INFO Persisting configuration
2017-05-17 20:43:49      scitran.api                  base.py   427:DEBU from local None POST /api/login {} request_id=4f622ea6-1495053829
=========================================== 1 failed, 30 passed in 2.30 seconds ============================================
kill: usage: kill [-s sigspec | -n signum | -sigspec] pid | jobspec ... or kill -l [sigspec]
```

--- a/test/unit_tests/python/conftest.py
+++ b/test/unit_tests/python/conftest.py
@@ -44,7 +44,10 @@ def log_db(app):
 
 @pytest.yield_fixture(scope='session')
 def app():
-    """Return api instance that uses mocked MongoClient"""
+    """Return api instance that uses mocked os.environ and MongoClient"""
+    env_patch = mock.patch.dict(
+        os.environ, {'SCITRAN_CORE_DRONE_SECRET': SCITRAN_CORE_DRONE_SECRET}, clear=True)
+    env_patch.start()
     mongo_patch = mock.patch('pymongo.MongoClient', new=mongomock.MongoClient)
     mongo_patch.start()
     # NOTE db and log_db is created at import time in api.config
@@ -52,6 +55,7 @@ def app():
     reload(api.config)
     yield api.web.start.app_factory()
     mongo_patch.stop()
+    env_patch.stop()
 
 
 @pytest.fixture(scope='session')

